sql.go

```diff
@@ -313,9 +313,11 @@ var migrations = []migration{
 type Storage struct {
 	db *sql.DB
 
-	// go-sqlite supports concurrent reads, but not writes. Queries that write
-	// should use this mutex to synchronize that access
-	writeMu sync.Mutex
+	// sqlite supports either one writer, or multiple readers. wrap tables in
```

Comment:
TIL

Reply:
I'm still not 100% sure where the concurrency boundary is, I can find a lot of takes on various threads and issues but no real succinct proof.

I did manage to confirm that an insert while a select is running does fail with a table lock tho.

Reply:
Aaah this is a little insightful: https://github.com/mattn/go-sqlite3/issues/632#issue-360437494

Seems like it's not the database concurrency in https://sqlite.org/lockingv3.html we're hitting, it's actually queries being interleaved on the same connection.

After reading through that it seems like the shared cache is not really making things better, and a busy timeout would be a more appropriate solution. Gonna give that a shot.